src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -2087,12 +2087,14 @@ private bool pInvokeMarshalingRequired(CORINFO_METHOD_STRUCT_* handle, CORINFO_S
                     return true;
                 }
 
-                return ((PInvokeILStubMethodIL)stubIL).IsMarshallingRequired;
+                PInvokeILStubMethodIL pinvokeILStub = (PInvokeILStubMethodIL)stubIL;
+                return pinvokeILStub.IsMarshallingRequired || pinvokeILStub.HasByRefArguments;
```

Comment:
I do not think it makes sense to be changing the meaning of `IsMarshalingRequired`. We have always considered pinning of byrefs to be one kind of marshaling and we should keep it that way.